src/main/java/jarvis/model/LessonBook.java

```diff
@@ -0,0 +1,112 @@
+package jarvis.model;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+
+import javafx.collections.ObservableList;
+
+/**
+ * Wraps all lesson data in JARVIS
+ * Duplicates are not allowed (by .equal comparison)
+ */
+public class LessonBook implements ReadOnlyLessonBook {
+
+    private final UniqueLessonList lessons;
+
+    {
+        lessons = new UniqueLessonList();
+    }
+
+    public LessonBook() {}
+
+    /**
+     * Creates a LessonBook using the Lessons in the {@code toBeCopied}
+     */
+    public LessonBook(ReadOnlyLessonBook toBeCopied) {
+        this();
+        resetData(toBeCopied);
+    }
+
+    //// list overwrite operations
+
+    /**
+     * Replaces the contents of the lesson list with {@code lessons}.
+     * {@code lessons} must not contain duplicate lessons.
+     */
+    public void setLessons(List<Lesson> lessons) {
+        this.lessons.setLessons(lessons);
+    }
+
+    /**
+     * Resets the existing data of this {@code LessonBook} with {@code newData}.
+     */
+    public void resetData(ReadOnlyLessonBook newData) {
+        requireNonNull(newData);
+
+        setLessons(newData.getLessonList());
+    }
+
+    //// lesson-level operations
+
+    /**
+     * Returns true if a lesson with the same identity as {@code lesson} exists in the lesson book.
+     */
+    public boolean hasLesson(Lesson lesson) {
+        requireNonNull(lesson);
+        return lessons.contains(lesson);
+    }
+
+    /**
+     * Adds a lesson to the lesson book.
+     * The lesson must not already exist in the lesson book.
+     */
+    public void addLesson(Lesson p) {
+        lessons.add(p);
```

Comment:
Can consider iterating through the current lesson list to check if there are any timing conflicts using `Lesson::hasTimingConflict`